### PR TITLE
SamlPostProfileResponse Endpoint enhancements

### PR DESCRIPTION
**Pull request by tsschmidt**
PR adds the option to send parameters to the endpoint as a map using the http Post.

Adds 'encrypt' as parameter.  The default is to now return an unencrypted response and attribute statement.  If set to true the response and the attribute statement will be encrypted.

Sets the formatting of the response to be indented.

### CODE REVIEW

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java [hunk 1]
```diff
@@ -95,22 +101,57 @@ public SSOSamlIdPPostProfileHandlerEndpoint(final CasConfigurationProperties cas
...
+     * @param map the RequestBody
+     * @return the response entity
+     */
+    @PostMapping(produces = MediaType.APPLICATION_XML_VALUE)
+    @ResponseBody
+    public ResponseEntity<Object> producePost(final HttpServletRequest request,
+                                              final HttpServletResponse response,
+                                              final @RequestBody Map<String, String> map) {
+
+        val username = map.get("username");
```
Comment (mmoayyed) [inline support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: Not sure I follow this; if you mean to support both GET and POST, a RequestMapping on the original method with both methods should do the job, and would be less code.

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java [hunk 2]
```diff
@@ -95,22 +101,57 @@ public SSOSamlIdPPostProfileHandlerEndpoint(final CasConfigurationProperties cas
...
         try {
-            val username = request.getParameter("username");
-            val password = request.getParameter("password");
-            val entityId = request.getParameter(SamlProtocolConstants.PARAMETER_ENTITY_ID);
-            
             val selectedService = this.serviceFactory.createService(entityId);
             val registeredService = this.servicesManager.findServiceBy(selectedService, SamlRegisteredService.class);
             RegisteredServiceAccessStrategyUtils.ensureServiceAccessIsAllowed(registeredService);
 
+            val loadedService = (SamlRegisteredService) JSON_SERIALIZER.from(JSON_SERIALIZER.toString(registeredService));
```
Comment (mmoayyed) [inline support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:146]: You already have a registered-service; why serialize it to JSON, only to deserialize it again, to get the object again?

**Review by mmoayyed — CHANGES_REQUESTED** [review]

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 1 above)
Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: Hmm, How do you get parameters out of the RequestBody, compared to getting them as url params?

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 2 above)
Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:146]: Setting the fields on the object directly from the ServicesManager persists the change in the stored object.  Serializing to a new object ensures that the functionality is not affected for actual requests for the service.

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 1 above)
Comment (mmoayyed) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: I think the method signature should just be using @RequestParam, and then Spring will take care of it.

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 2 above)
Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:146]: I think I tried that, but clone() alone wasn't deep enough.  I can you point me to a util or something.  Serializing to new object seemed the cleanest.

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 1 above)
Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: I will give it a try

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 2 above)
Comment (mmoayyed) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:146]: How about `BeanUtils.copyProperties(..., ...);` ?

File: support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java (hunk 1 above)
Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: yes, content header is application/json and sending the request body as a JSON string

Comment (mmoayyed) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: Most likely, the method here should indicated via "consumes" that it accepts JSON. That's likely where the bad-request comes from. Can you share a sample JSON payload with me?

Comment (tsschmidt) [reply support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java:127]: No worries, always good to explore all options.

**Pull request merged**